CONTRIBUTING.md

```diff
@@ -0,0 +1,233 @@
+# Contributing to VFS for Git
+
+Thank you for taking the time to contribute!
+
+## Guidelines
+
+* [Code of Conduct](#code-of-conduct)
+* [Design Reviews](#design-reviews)
+* [Platform Specific Code](#platform-specific-code)
+* [Tracing and Logging](#tracing-and-logging)
+* [Error Handling](#error-handling)
+* [Background Threads](#background-threads)
+* [Coding Conventions](#coding-conventions)
+* [Testing](#testing)
+
+## Code of Conduct
+
+This project has adopted the [Microsoft Open Source Code of
+Conduct](https://opensource.microsoft.com/codeofconduct/).
+For more information see the [Code of Conduct
+FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or
+contact [opencode@microsoft.com](mailto:opencode@microsoft.com)
+with any additional questions or comments.
+
+## Design Reviews
+
+Architectural changes and new features should start with a design review.  It's easier and wastes less time to incorporate feedback at this stage.
+
+The design review process is as follows:
+
+1. Create a pull request that contains a design document in Markdown (.md) format for the proposed change.  Assign the `design-doc` label to the pull request.
+2. Use the pull request for design feedback and for iterating on the design.
+3. Once the design is approved, create a new issue with a description that includes the final design document.  Include a link to the pull request that was used for discussion.
+4. Close (without merging!) the pull request used for the design discussion.
+
+## Platform Specific Code
+
+- *Prefer cross-platform code to platform-specific code*
+
+  Cross-platform code is more easily reused.  Reusing code reduces the amount of code that must be written, tested, and maintained.
+
+- *Platform specific code, and only platform specific code, should go in `GVFSPlatform`*
+
+  When platform specific code is required, it should be placed in `GVFSPlatform` or one of the platforms it contains (e.g. `IKernelDriver`)
+
+## Tracing and Logging
+
+- *The "Error" logging level is reserved for non-retryable errors that result in I/O failures or the VFS4G process shutting down*
```

Comment:
This seems to be a common issue that people stumble over. When an "unexpected condition" is encountered (exception, or other problem), it seems people use the Error helper methods. Maybe we could do something to make it easier for developers do the right thing here.

Reply:
I completely agree, I will file a work item for this.

Reply:
Filed #895 for this